core/codec/common.hpp

```diff
@@ -15,8 +16,8 @@ namespace fc::codec {
       out = {};
       return false;
     }
-    out = input.subspan(0, n);
-    input = input.subspan(n);
+    out = input.subspan(0, static_cast<gsl::span<BytesIn>::index_type>(n));
+    input = input.subspan(static_cast<gsl::span<BytesIn>::index_type>(n));
```

Comment:
```suggestion
    out = input.subspan(0, static_cast<ptrdiff_t>(n));
    input = input.subspan(static_cast<ptrdiff_t>(n));
```
or could have been "BytesIn::index_type" aka "gsl::span<uint8_t>::index_type"